### PR TITLE
Security vulnerabilities database

### PR DESCRIPTION
**Pull request by Cheshiriks**
* Added FossGraph view for security vulnerabilities database

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+/**
+ * [VFC] for foss graph view
+ */
+@Suppress("EMPTY_BLOCK_STRUCTURE_ERROR")
+val fossGraphView: VFC = VFC {
+    val fossGraphView = fossGraphView()
+    fossGraphView {
+
+    }
+}
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt:41]: What should this code lines do? It seems to be wrong

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                            ariaLabel = "Search"
+                            ariaDescribedBy = "basic-addon2"
+                        }
+                        div {
+                            className = ClassName("input-group-append")
+                            button {
+                                className = ClassName("btn btn-primary")
+                                type = ButtonType.button
+                                fontAwesomeIcon(icon = faSearch, classes = "trash-alt")
+                            }
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt:82]: Use `buttonBuilder` instead

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                        }
+                    }
+                }
+            }
+        }
+
+        h1 {
+            className = ClassName("h3 mb-0 text-center text-gray-800")
+            // +(props.name)
+            +"CVE-2022-22978"
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt:92]: ?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                                    display = Display.flex
+                                    alignItems = AlignItems.center
+                                }
+                                +"Description"
+                            }
+                        }
+                    }
+                    div {
+                        className = ClassName("card-body")
+                        ReactHTML.textarea {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt:132]:
```suggestion
                        textarea {
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt:92]: This is just a temporary stub for an example.

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/topbar/TopBar.kt [hunk 5]
```diff
@@ -26,11 +26,6 @@ external interface TopBarProps : PropsWithChildren {
      * Currently logged in user, or `null`.
      */
     var userInfo: UserInfo?
-
-    /**
-     * `true`, if the device is mobile (screen is less 1000px).
-     */
-    var isMobileScreen: Boolean?
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/topbar/TopBar.kt:33]: It has never been used. Probably once added and forgotten.

**Review by nulls — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+     */
+    var progress: Int
+}
+
+@Suppress(
+    "MAGIC_NUMBER",
+    "TOO_LONG_FUNCTION",
+    "LongMethod",
+)
+private fun fossGraphView(): FC<FossGraphViewProps> = FC { props ->
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/FossGraphView.kt:65]: Suggestion to keep it `val` and maybe create a temp wrapper of type `VFC`. Just like you have now but without line `38`

**Review by sanyavertolet — APPROVED** [review]
Update screenshot, jic

**Pull request merged**